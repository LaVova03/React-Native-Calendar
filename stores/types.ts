import { EventFormData } from "@/types/types";

export interface StoreState {
  isModal: {
    form: boolean;
    edit: boolean;
    confirm: boolean;
  };
  data: EventFormData[];
  setData: (events: EventFormData[]) => void;
  setIsModal: (key: keyof StoreState["isModal"], value: boolean) => void;
}
