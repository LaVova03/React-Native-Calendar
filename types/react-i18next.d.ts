import "react-i18next";

declare module "react-i18next" {
  interface Resources {
    en: {
      translation: typeof import("./locales/en.json");
    };
    uk: {
      translation: typeof import("./locales/uk.json");
    };
  }
}
