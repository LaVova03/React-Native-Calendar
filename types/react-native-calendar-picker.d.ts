declare module "react-native-calendar-picker" {
  import * as React from "react";
  import { ViewStyle, TextStyle } from "react-native";

  export interface CalendarPickerProps {
    onDateChange: (date: any) => void;
    selectedDate?: string | Date;
    todayBackgroundColor?: string;
    selectedDayColor?: string;
    selectedDayTextColor?: string;
    headerWrapperStyle?: ViewStyle;
    headerStyle?: ViewStyle;
    monthTitleStyle?: TextStyle;
    dateTextStyle?: TextStyle;
    customDatesStyles?: Array<{
      date: Date;
      style: ViewStyle;
      textStyle: TextStyle;
    }>;
    previousComponent?: React.ReactNode;
    nextComponent?: React.ReactNode;
  }

  export default class CalendarPicker extends React.Component<CalendarPickerProps> {}
}
