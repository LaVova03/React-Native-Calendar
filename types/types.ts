import { Moment } from "moment";
import React from "react";

export interface EventFormData {
  eventName: string;
  startDate: string;
  startTime: string;
  repeat: string;
  id?: number;
}

export interface EventCardProps {
  event: EventFormData;
  setIdEvent?: (value: number | null) => void;
  setPutevent?: (value: EventFormData) => void;
}

export interface FormProps {
  selectedDate?: Moment | null;
  events?: Event[];
  setEvents?: React.Dispatch<React.SetStateAction<Event[]>>;
  editEvent?: EventFormData | null;
}

export interface SelectProps {
  setFormData: React.Dispatch<React.SetStateAction<EventFormData>>;
  formData: EventFormData;
}

export interface ModalFormProps {
  children: React.ReactNode;
}

export interface ConfirmProps {
  setConfirmDelete: React.Dispatch<React.SetStateAction<boolean>>;
}
