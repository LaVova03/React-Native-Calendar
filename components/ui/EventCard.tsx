import { EventCardProps, EventFormData } from "@/types/types";
import React from "react";
import { Pressable, StyleSheet, Text, View } from "react-native";
import Ionicons from "react-native-vector-icons/Ionicons";
import { useStore } from "@/stores/useStore";
import moment from "moment";
import { showAlert } from "@/utils/alert";

function EventCard({ event, setIdEvent, setPutevent }: EventCardProps) {
  const { setIsModal } = useStore();

  const editEvent = (event: EventFormData) => {
    const eventDate = moment(event.startDate, "MMM D.YYYY");
    const today = moment().startOf("day");

    if (eventDate.isBefore(today)) {
      showAlert("Error", "You cannot edit an event in the past!");
      return;
    }
    setIsModal("edit", true);
    if (event.id && setPutevent) setPutevent(event);
  };
  return (
    <View style={styles.container}>
      <View>
        <Text style={styles.eventName}>{event.eventName}</Text>
      </View>
      <View style={styles.row}>
        <Text style={styles.item}>{event.startDate}</Text>
        <Text style={styles.item}>{event.startTime}</Text>
        <Text style={styles.item}>{event.repeat}</Text>
      </View>
      <View style={styles.btnCont}>
        <Pressable
          onPress={() => {
            setIsModal("confirm", true);
            if (event.id && setIdEvent) setIdEvent(event.id);
          }}
        >
          <Ionicons
            name="trash-bin"
            size={20}
            color="black"
            style={styles.icon}
          />
        </Pressable>
        <Pressable
          onPress={() => {
            editEvent(event);
          }}
        >
          <Ionicons name="create" size={20} color="black" style={styles.icon} />
        </Pressable>
      </View>
    </View>
  );
}

export default EventCard;

const styles = StyleSheet.create({
  container: {
    width: "100%",
    height: "auto",
    paddingHorizontal: 20,
    paddingVertical: 10,
    borderWidth: 1,
    borderColor: "black",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.3,
    shadowRadius: 4,
    shadowColor: "rgb(255, 213, 0)",
    borderRadius: 20,
    gap: 20,
  },
  eventName: {
    width: "auto",
    textAlign: "center",
    borderBottomWidth: 1,
    borderBottomColor: "black",
    paddingBottom: 5,
    fontSize: 20,
    fontWeight: 600,
  },
  row: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
  },
  btnCont: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    gap: 20,
  },
  icon: {
    backgroundColor: "rgb(255 213 0)",
    borderRadius: 15,
    padding: 5,
  },
  item: {
    width: "30%",
    textAlign: "center",
    borderBottomWidth: 1,
    borderBottomColor: "black",
    paddingBottom: 5,
    fontWeight: 600,
  },
});
