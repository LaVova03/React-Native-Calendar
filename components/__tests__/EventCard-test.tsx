import * as React from "react";
import renderer from "react-test-renderer";
import EventCard from "@/components/ui/EventCard";
import { EventFormData } from "@/types/types";

jest.mock("@/stores/useStore", () => ({
  useStore: jest.fn(() => ({ setIsModal: jest.fn() })),
}));

jest.mock("react-native-vector-icons/Ionicons", () => "Ionicons");

const mockSetIdEvent = jest.fn();
const mockSetPutevent = jest.fn();

const event: EventFormData = {
  eventName: "Test Event",
  startDate: "Jan 1.2025",
  startTime: "10:00 AM",
  repeat: "None",
};

describe("EventCard", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <EventCard
          event={event}
          setIdEvent={mockSetIdEvent}
          setPutevent={mockSetPutevent}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
