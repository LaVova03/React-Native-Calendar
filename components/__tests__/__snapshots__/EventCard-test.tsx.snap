// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard renders correctly 1`] = `
<View
  style={
    {
      "borderColor": "black",
      "borderRadius": 20,
      "borderWidth": 1,
      "gap": 20,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
      "shadowColor": "rgb(255, 213, 0)",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 4,
      "width": "100%",
    }
  }
>
  <View>
    <Text
      style={
        {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "fontSize": 20,
          "fontWeight": 600,
          "paddingBottom": 5,
          "textAlign": "center",
          "width": "auto",
        }
      }
    >
      Test Event
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "fontWeight": 600,
          "paddingBottom": 5,
          "textAlign": "center",
          "width": "30%",
        }
      }
    >
      Jan 1.2025
    </Text>
    <Text
      style={
        {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "fontWeight": 600,
          "paddingBottom": 5,
          "textAlign": "center",
          "width": "30%",
        }
      }
    >
      10:00 AM
    </Text>
    <Text
      style={
        {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "fontWeight": 600,
          "paddingBottom": 5,
          "textAlign": "center",
          "width": "30%",
        }
      }
    >
      None
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 20,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Ionicons
        color="black"
        name="trash-bin"
        size={20}
        style={
          {
            "backgroundColor": "rgb(255 213 0)",
            "borderRadius": 15,
            "padding": 5,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Ionicons
        color="black"
        name="create"
        size={20}
        style={
          {
            "backgroundColor": "rgb(255 213 0)",
            "borderRadius": 15,
            "padding": 5,
          }
        }
      />
    </View>
  </View>
</View>
`;
